packages/marketplace-ui/src/components/lbc/mint/candy-machine.ts

```diff
@@ -268,103 +302,55 @@ export const mintOneToken = async (
         )[0],
         isWritable: false,
         isSigner: false,
-      });
+      })
     }
   }
   if (candyMachine.whitelistMintSettings) {
     const mint = new anchor.web3.PublicKey(
       candyMachine.whitelistMintSettings.mint
-    );
+    )
 
-    const whitelistToken = (await getAtaForMint(mint, payer))[0];
+    const whitelistToken = (await getAtaForMint(mint, payer))[0]
     remainingAccounts.push({
       pubkey: whitelistToken,
       isWritable: true,
       isSigner: false,
-    });
+    })
 
     if (candyMachine.whitelistMintSettings.mode.burnEveryTime) {
-      const whitelistBurnAuthority = anchor.web3.Keypair.generate();
-
       remainingAccounts.push({
         pubkey: mint,
         isWritable: true,
         isSigner: false,
-      });
+      })
       remainingAccounts.push({
-        pubkey: whitelistBurnAuthority.publicKey,
+        pubkey: payer,
         isWritable: false,
         isSigner: true,
-      });
-      signers.push(whitelistBurnAuthority);
-      const exists =
-        await candyMachine.program.provider.connection.getAccountInfo(
-          whitelistToken
-        );
-      if (exists) {
-        instructions.push(
-          Token.createApproveInstruction(
-            TOKEN_PROGRAM_ID,
-            whitelistToken,
-            whitelistBurnAuthority.publicKey,
-            payer,
-            [],
-            1
-          )
-        );
-        cleanupInstructions.push(
```

Comment:
It's not needed as we switch to using the token ATA rather than `createApproveInstruction`.
New logic is here: https://github.com/metaplex-foundation/metaplex/blob/e20c14069be57ffa09428dd25a07a5bcf7ef51c4/js/packages/candy-machine-ui/src/candy-machine.ts#L448

Reply:
Nice